controllers/TasksController.php

```diff
@@ -3,37 +3,95 @@
 namespace app\controllers;
 
 use Yii;
+use yii\filters\AccessControl;
 use yii\web\Controller;
 use yii\web\NotFoundHttpException;
+use yii\web\Response;
+use yii\web\UploadedFile;
+use yii\widgets\ActiveForm;
+use app\models\forms\AddTaskForm;
 use app\models\forms\SearchForm;
 use app\services\TaskService;
 use app\services\CategoryService;
+use app\services\UserService;
 
 class TasksController extends SecuredController
 {
+    public function behaviors()
+    {
+        return [
+            'access' => [
+                'class' => AccessControl::class,
+                'rules' => [
+                    [
+                        'allow' => true,
+                        'roles' => ['@'],
+                        'actions' => ['create'],
+                        'matchCallback' => function ($rule, $action) {
+                            return (new UserService)->isCustomer(Yii::$app->user->id);
+                        }
+                    ],
+                    [
+                        'allow' => true,
+                        'roles' => ['@'],
+                        'actions' => ['index', 'view']
+                    ]
+                ]
+            ]
+        ];
+    }
+
+    public function actionCreate()
+    {
+        $addTaskForm = new AddTaskForm;
+
+        if (Yii::$app->request->isPost) {
+            $addTaskForm->load(Yii::$app->request->post());
+            $addTaskForm->files = UploadedFile::getInstances($addTaskForm, 'files');
+
+            if (Yii::$app->request->isAjax) {
+                Yii::$app->response->format = Response::FORMAT_JSON;
+
+                return ActiveForm::validate($addTaskForm);
+            }
+
+            if ($addTaskForm->validate()) {
+                $taskId = (new TaskService)->create($addTaskForm);
+                $this->redirect(['tasks/view', 'id' => $taskId]);
+            }
+        }
+
+        $categories = (new CategoryService)->getAllCategories();
+
+        return $this->render('add', [
+            'model' => $addTaskForm,
+            'categories' => $categories
+        ]);
+    }
+
     public function actionIndex(?string $category = null)
     {
-        $model = new SearchForm();
+        $searchForm = new SearchForm;
```

Comment:
скобочки в конце лучше ставить
```suggestion
        $searchForm = new SearchForm();
```